[Nitant Panicker] iP

nitant-p opened the pull request:
# Ploopy

> "It's not a bug - it's a feature" - Someone funny

Ploopy lets you handle your todo list so you don't have to remember all the important stuff!

It's: 

- very easy to use
- not that fast (yet)
- ~~pretty funny~~ tolerable

Get started:

1. Download it [here](https://github.com/nitant-p/ip/releases/tag/v0.1)
2. Open it
3. Follow the instructions and add your stuff! 😎 

Best of all it's **FREEEEEEEE**

We can :

- [x] Remember your tasks
- [x] Forget your tasks
- [x] Find your tasks
- [ ] Remind you of deadlines

If you're a Java programmer, you can use it to practice Java too. Here's our `main` method:

```java

public static void main(String[] args) {
        new Ploopy().start();
    }

```

Commit:
Add Gradle support

nitant-p committed:
Created Greet, Echo and Exit capabilities

nitant-p committed:
no message

nitant-p committed:
Added list capabilities.

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Revert "no message"

This reverts commit dd52ec7c14f486c1f18f5c109d3b0ef0d4ceba1c.

nitant-p committed:
Added mark and unmark task. Improved OOP?

nitant-p committed:
Added 3 Tasks: ToDo, Event, Deadline. Made Task abstract

nitant-p committed:
Finished all 3 event types

nitant-p committed:
Added Automated Testing. Actual matches with Expected for now...

nitant-p committed:
Added Exceptions to be created, thrown and caught.

nitant-p committed:
Added Delete Function; Removed DukeExcpetion child classes.

nitant-p committed:
Added missing markTask method.

nitant-p committed:
no message

nitant-p committed:
Add save to hard drive

nitant-p committed:
Change date format from String to LocalDateTime; Support different formats and time

nitant-p committed:
Merge branch 'branch-Level-7'

nitant-p committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/ToDo.java

nitant-p committed:
Add more classes and improve OOP

nitant-p committed:
Package all classes into duke package

nitant-p committed:
Package all classes into package ploopy

nitant-p committed:
Merge branch 'add-gradle-support'

nitant-p committed:
Add Gradle support

nitant-p committed:
Add Gradle Support to build and run

nitant-p committed:
Add JUnit Tests for ToDo, Deadline and Parser

nitant-p committed:
Add JavaDoc comments.

nitant-p committed:
Add Find Task capability

nitant-p committed:
Use checkstyle and correct to coding standards

nitant-p committed:
Merge branch 'branch-A-JavaDoc'

nitant-p committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/ploopy/UI.java

nitant-p committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ploopy/Parser.java
#	src/main/java/ploopy/TaskList.java

nitant-p committed:
Add format prompt

nitant-p committed:
Improve date parser

nitant-p committed:
Remove case sensitivity

nitant-p committed:
Fix coding standard violations using checkstyle

wongyewjon commented on src/main/java/ploopy/UI.java:
```diff
@@ -0,0 +1,172 @@
+ * Interacts with the user between commands
+ *
+ */
+public class UI {
```
i like how detailed the UI is 👍

wongyewjon commented on src/main/java/ploopy/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+package ploopy;
+
+public class Deadline extends Task {
```
Perhaps you can have separation of the files into different packages as you deem fit.

wongyewjon reviewed (commented):
Looks super good. 
Really clean and neat code, made reading the implementation really enjoyable.
Thanks, I have learnt a lot.

Vshnv2001 commented on src/main/java/ploopy/Task.java:
```diff
@@ -0,0 +1,135 @@
+ *
+ */
+
+public abstract class Task {
```
I like your use of inheritance by abstracting common methods of Events, Deadlines and ToDo into a class Task!

Vshnv2001 commented on src/main/java/ploopy/Task.java:
```diff
@@ -0,0 +1,135 @@
+     *
+     * @return Task done status.
+     */
+    public boolean isDone() {
```
I like how you have adhered to the coding standard in naming methods that return a boolean value

Vshnv2001 commented on src/main/java/ploopy/Task.java:
```diff
@@ -0,0 +1,135 @@
+     *
+     */
+    public void markDone() {
+        done = true;
```
Perhaps you could use this.done since it is an attribute?

Vshnv2001 commented on src/main/java/ploopy/PloopyException.java:
```diff
@@ -0,0 +1,14 @@
+ * An exception class that encapsulates the error message during
+ * the execution of the program.
+ */
+public class PloopyException extends Exception {
```
Perhaps you could make your exception name more descriptive, as to what kind of an error it is?

ama-chi commented on src/main/java/ploopy/PloopyException.java:
```diff
@@ -0,0 +1,14 @@
+ */
+public class PloopyException extends Exception {
+
+    PloopyException(String message) {
```
Could consider making child-classes of PloopyException to further specify the type of error.

ama-chi commented on src/main/java/ploopy/Task.java:
```diff
@@ -0,0 +1,135 @@
+
+public abstract class Task {
+    /** Input string to Date formatter */
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy kkmm");
+    /** Date to output String formatter */
+    private static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy kkmm");
```
I like how you extracted the format into variables here.

ama-chi commented on src/main/java/ploopy/TaskList.java:
```diff
@@ -0,0 +1,172 @@
+    public void createEvent(String name, String date) throws PloopyException {
+        Task newTask = new Event(name, date);
+        taskList.add(newTask);
+        ui.addTaskMessage(newTask, taskList.size());
+        storage.writeToFile(newTask);
```
This code seems to be repeated multiple times in this class, perhaps there is a better way to structure this.

ama-chi reviewed (commented):
Overall, code is relatively clean and inheritance is well used here, good job!

nitant-p committed:
no message

nitant-p committed:
Fix coding standards violations using checkstyle script

nitant-p committed:
Add GUI and remove text UI

nitant-p committed:
Merge branch 'branch-Level-10'

nitant-p committed:
Merge branch 'branch-A-Checkstyle'

nitant-p committed:
Add assertions for Parser methods

At the beginning of the parse() method, there is a check for if the
input is empty or blank. An assertion was added for this to ensure
the private isIncompleteMethod() actually receives a non-empty String.

An assertion was added for getDate() to check if the date was correctly
extracted (for the specified format).

An assertion was added in parse() to check if isIncompleteMethod()
works correctly.

nitant-p committed:
Merge pull request #1 from nitant-p/branch-A-Assertions

Add assertions for Parser methods

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

* 'master' of https://github.com/nitant-p/ip:
  Add assertions for Parser methods

nitant-p committed:
Improve coding quality

nitant-p committed:
Improve coding quality

Removed repeated lines of code in parse() method and added new methods
to parse tasks and commands.

Removed individual task creation methods in TaskList and unified them
into one task method.

nitant-p committed:
Merge pull request #2 from nitant-p/branch-A-CodeQuality

Improve coding quality

nitant-p committed:
Add Priority for Tasks

To allow the user to set priorities for their tasks.

Add a priority variable to the Task class and create setters and
getters.

This makes it easy as the rest of the task classes can have the same
priority variable via inheritance.

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Merge branch 'branch-BCD-Extension'

nitant-p committed:
Package task classes into task package

Change toString() methods for all classes

nitant-p committed:
Update README.md

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Update README.md

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Use non-copyrighted images for Ploopy and User images

nitant-p committed:
Fix checkstyle violations

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Add Ui.png

nitant-p committed:
no message

nitant-p committed:
Update README.md

nitant-p committed:
Change Ui.png and add title to GUI window

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Change javafx version in fxml file

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md

nitant-p committed:
Merge branch 'master' of https://github.com/nitant-p/ip

nitant-p committed:
Support incorrect input format for event and date

nitant-p committed:
no message

nitant-p committed:
Update README.md

nitant-p committed:
Update README.md
